### DISCUSSION TITLE
12V Battery SoC no longer in JSON?

### DISCUSSION THREAD

**Discussion by binderth** [Hacksore/bluelinky#215] [Q&A]
As of beginning of May I don't receive the 12V SoC anymore as it seems. Is it something European Users exerience also?
as described here: https://github.com/Hacksore/bluelinky/issues/212 I got the `vehicleStatus.battery.batSoc` but now it's gone?

on the other hand, I'm experiencing a dead 12V battery and it's loading from the 800V battery right now and `vehicleStatus.battery.batState` is `1` right now, opposed to `0` by the time I got a correct answer.

anyone else getting no more 12V SoC? or is it just me...?

on the other hand, it seems, that since 7th of May, there's no more changing in 12V SoC value and from the way I implemented the readout (via NodeRed), it _could_ be, that it sent the last SoC (93) over and over again:
![grafik](https://user-images.githubusercontent.com/9037329/170971984-bb2a1a05-d4f6-484e-853c-7b0d13638284.png)

**Comment by PierreLevres** [comment]
My 12V SoC is still being updated...

**Reply by binderth** [reply]
hmm... ok. Then I'll have to check, when the KIA dealer is done with my car. The battery went dead yesterday and it had to be towed to the dealership

**Comment by PierreLevres** [comment]
Good luck.  My battery was replaced also, just before the end of the two year warranty period. Got a jumpstart from the roadside assistance who concluded the battery was in bad shape.

**Reply by binderth** [reply]
Thanks. Happened twice now. In March it was, because I was PV excess loading over the weekend - and obviously KIA/Hyundai has issues with open loading flaps or some sorts. That was one of the reasons I monitored the 12V SoC in the first place, to get alerted if too low.
But as it died yesterday morning, I saw it stayed on 93 the last three weeks -> which is because I ticked "only new values" in Node-Red and a NULL-Value obviously doesn't count as "new"... So again some kind of controlling device still sucks the 12V battery dry.

Now the car is on full testing at the dealer's.

**Comment by binderth** [comment]
btw: just realised, that I'm missing the 12V Battery icon in the App also. There was on in "Others", it is missing now.
![grafik](https://user-images.githubusercontent.com/9037329/171120239-dc1e948a-d578-46d5-8365-c28fbd68b516.png)

**Comment by PierreLevres** [comment]
Well the good news is that your example shows that bluelinky perfectly mimics the app. 

Your battery is faulty but also the sensor might be broken in your car. Roadside assistance told me that jumpstarting can break that sensor… 

The 12v battery status is only send, even if you pull the car actively, when you turn off the car. That’s why I decided not to actively pull the cars information anymore. Too risky for my use case. I solely rely on the car pushing info to the Kia servers when it is shut off, and that I query every 15 minutes.

**Reply by binderth** [reply]
That's strange, that jumpstarting will break the battery sensor... But ok, you'll never know.

What I did since March was to use `FULL` cached-results every 15mins and get a `FULL` refresh-result every 2hours. Worked fine - right until yesterday. On the weekend of May 7th we drove the car 220km two days before, it was "fast-charged" (50kW) on the 6th and on the 7th it stood parking open air for the day. Only in the evening we drove like 3km and that's where the 93% SoC sticked and presumably the 12V battery sensor was lost. Let's see, what the dealer can accomplish.

**Reply by binderth** [reply]
Neither in my JSON-queries via bluelinky (`FULL` *and* `CACHED`) the `vehicleStatus.battery.batSoc` is present, nor in my App is the Battery-Icon visible. 

My car is back from the dealer's and the 12V battery was obviously defective and was replaced. but still, there's no new info on the 12V SoC.

**Comment by PierreLevres** [comment]
My battsoc wasnt visible after my battery was replaced, but eventually it got back again. Maybe after the first charge, maybe after a reset of the system, or an update…
Maybe be patient for a day or two and then call Kia or the dealer? It should turn up in your app… (and then it will in bluelinky too).

**Reply by binderth** [reply]
I don't think, calling KIA directly will do anything! ;)
I do think, they either have a bug in their API/car computer - or [putting on my alu tin hat] they don't want to expose the battery-icon in the App for cars which could have the parking brake error! 😁

**Comment by PierreLevres** [comment]
Any progress?
Did they really replace the sensor?

https://www.samarins.com/glossary/battery-sensor.html
